test/modules/DirectMethodSender/ReporterClients/EventReporterClient.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace DirectMethodSender
+{
+    using System;
+    using System.Net;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    internal sealed class EventReporterClient : ReporterClientBase
+    {
+        ILogger logger;
+        TransportType transportType;
+        ModuleClient moduleClient = null;
+        ModuleClient ModuleClient
+        {
+            get
+            {
+                if (this.moduleClient == null)
+                {
+                    Preconditions.CheckNotNull(this.transportType, nameof(this.transportType));
+                    this.moduleClient = ModuleUtil.CreateModuleClientAsync(
+                        this.transportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        this.logger).Result;
+                    return this.moduleClient;
+                }
+                else
+                {
+                    return this.moduleClient;
+                }
+            }
+        }
+
+        public EventReporterClient(ILogger logger)
+            : base(logger)
+        {
+            this.logger = Preconditions.CheckNotNull(logger, nameof(logger));
+        }
+
+        public override void Dispose()
+        {
+            this.moduleClient?.Dispose();
+        }
+
+        internal async Task<EventReporterClient> InitAsync(TransportType transportType)
```

Comment:
please wrap this Init logic inside a private module client property or a private function or lazy type.

Reply:
it does. See the `ModuleClient`